biosys/apps/main/tests/api/test_schema_inference.py

```diff
@@ -260,6 +262,90 @@ def test_mix_types_infer_most_plausible(self):
             field = schema.get_field_by_name('How Many')
             self.assertEquals(field.type, 'integer')
 
+    def test_csv_with_excel_content_type(self):
+        """
+        Often on Windows a csv file comes with an excel content-type (e.g: 'application/vnd.ms-excel')
+        Test that we handle the case.
+        """
+        # In order to hack the Content-Type of the multipart form data we need to use the APIRequestFactory and work
+        # with the view directly. Can't use the classic API client.
+        from rest_framework.authtoken.models import Token
+        from rest_framework.test import APIRequestFactory
+        from main.api.views import InferDatasetView
+        from rest_framework.test import force_authenticate
```

Comment:
Do theses statements need to be within the function?

Reply:
with comment

Reply:
Pep 8 recommends putting all imports at the top of the module.

https://www.python.org/dev/peps/pep-0008/#imports

Only exception is if you have circular imports

Reply:
I know, I know, Mr. Pep8. I'll fix that.

Reply:
You so could have said Dr Pep8 (like Dr Pepper).